### PR TITLE
WIP: [DOCS] Rework documentation for python3 version of jupyter console

### LINKED ISSUE #59741: [DOCS] Developing New Modules makes reference to Python 2.x version of IPython
**Description**
The Developing New Modules documentation makes reference to installing and using the Python 2 version of IPython

**Suggested Fix**
Update to reference Python 3.x version of IPython as Python 2.x is now EOL

**Type of documentation**
Salt documentation

**Location or format of documentation**
https://docs.saltproject.io/en/latest/topics/development/modules/developing.html#interactive-debugging

### PR DESCRIPTION
**Pull request by ggiesen**
### What does this PR do?
Adjusts the documentation to use python3, and reflect that jupyter has been split from IPython

### What issues does this PR fix or reference?
Fixes: #59741

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs

### Commits signed with GPG?
No

### CODE REVIEW

File: doc/topics/development/modules/developing.rst [hunk 1]
```diff
@@ -48,8 +48,7 @@ a shell:
 
 .. code-block:: bash
 
-    # NOTE: use ipython2 instead of ipython for Arch Linux
-    ipython console --existing
+?    jupyter console --existing
```
Comment (krionbsd) [inline doc/topics/development/modules/developing.rst:51]: Is `?` a typo here?

Comment (ggiesen) [reply doc/topics/development/modules/developing.rst]: Yes, fixed in 546a859

File: doc/topics/development/modules/developing.rst [hunk 2]
```diff
@@ -48,8 +48,7 @@ a shell:
 
 .. code-block:: bash
 
-    # NOTE: use ipython2 instead of ipython for Arch Linux
-    ipython console --existing
+    jupyter console --existing
```
Comment (krionbsd) [inline doc/topics/development/modules/developing.rst:51]: Rereading this doc I still don't quite understand why `jupyter` install is required, could you elaborate a bit more?

**Review by krionbsd — CHANGES_REQUESTED** [review]

File: doc/topics/development/modules/developing.rst (hunk 2 above)
Comment (ggiesen) [reply doc/topics/development/modules/developing.rst:51]:
```
$ ipython3 console --existing
[TerminalIPythonApp] WARNING | Subcommand `ipython console` is deprecated and will be removed in future versions.
[TerminalIPythonApp] WARNING | You likely want to use `jupyter console` in the future
```

**Comment by krionbsd** [comment]
@ggiesen could you fix pre-commit?

**Review by twangboy — APPROVED** [review]

**Comment by Ch3LL** [comment]
@ggiesen is this still a WIP?

**Comment by Ch3LL** [comment]
Closing due to inactivity. Please let me know if you want me to re-open this PR or open a new PR against the master branch.

**Pull request closed without merging**